app/src/main/java/pl/gittobefit/generate_training/CardioTrainingPlan.java

```diff
@@ -0,0 +1,126 @@
+package pl.gittobefit.generate_training;
+
+import android.content.Context;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+import pl.gittobefit.database.AppDataBase;
+import pl.gittobefit.database.entity.training.Exercise;
+import pl.gittobefit.database.entity.training.SavedTraining;
+import pl.gittobefit.database.entity.training.WorkoutForm;
+import pl.gittobefit.database.pojo.ExerciseExecutionPOJODB;
+
+public class CardioTrainingPlan implements TrainingPlanGenerator
+{
+    private final String TRAINING_TYPE = "CARDIO";
+    private final static int SINGLE_STEP = 3;
+    private final static int MAX_RAND_TRIES = 100;
+    private WorkoutForm trainingForm;
+
+    private AppDataBase base;
+
+    public CardioTrainingPlan(Context context)
+    {
+        this.base = AppDataBase.getInstance(context);
+    }
+
+    @Override
+    public SavedTraining create(WorkoutForm trainingForm)
+    {
+        List<Integer> allExercisesId = base.exerciseDao().getAllByTrainingTypes_Name(TRAINING_TYPE);
```

Comment:
W sumie jeśli i tak musisz zaimplementować te metodę, to można jej zmienić na nazwę bez podłogi. Na serwerze Hibernate wymagał takiej składni.